staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/types.go

```diff
@@ -25,6 +25,9 @@ type CustomResourceDefinitionSpec struct {
 	// Group is the group this resource belongs in
 	Group string
 	// Version is the version this resource belongs in
+	// Should be always first item in Versions field if provided.
```

Comment:
to make the semantics clear, we can drop the `Version string` field from the internal types and only populate it during conversion.

Reply:
done

Reply:
Tried that, but we validate internal types and if it does not have version, then we need to do some validation in conversion path as the user facing type has version and versions. I prefer to keep this separated by keeping this Version field. I assume this is something internal that we can fix on a following PR if we want to.